library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -309,6 +313,46 @@ mkPrim prim ty = case prim of
     name <- Block.generateUniqueName "LitString"
     Block.globalString (toS s) name
 
+compileRecordDecl ::
+  Types.Define m =>
+  Monad m =>
+  Types.RecordDecl ->
+  -- | Primitive value.
+  Types.Annotated Types.TermLLVM ->
+  -- | Type of the primitive.
+  m LLVM.Operand
+compileRecordDecl (recordName, fieldDecls) term = do
+  Record.register recordName (map fst fieldDecls) (map (typeToLLVM . snd) fieldDecls)
+  compileTerm term
```

Comment:
beaitiful, deals with the complexity for us.

Should we make a Record type a "top level expression" so it doens't have to take any arguments.... I guess we could have both. This works

Reply:
Do you mean one with a dependent type?